packages/eslint-plugin/rules/i18n-translator-comments.js

```diff
@@ -0,0 +1,102 @@
+/**
+ * Internal dependencies
+ */
+const {
+	TRANSLATION_FUNCTIONS,
+	REGEXP_PLACEHOLDER,
+	getTranslateStrings,
+} = require( '../utils' );
+
+module.exports = {
+	meta: {
+		type: 'problem',
+		messages: {
+			missing:
+				'Translation function with placeholders is missing preceding translator comment',
+		},
+	},
+	create( context ) {
+		return {
+			CallExpression( node ) {
+				const {
+					callee,
+					loc: {
+						start: { line: currentLine },
+					},
+					parent,
+					arguments: args,
+				} = node;
+				if ( ! TRANSLATION_FUNCTIONS.includes( callee.name ) ) {
+					return;
+				}
+
+				const candidates = getTranslateStrings( callee.name, args );
+
+				if ( ! candidates ) {
+					return;
+				}
+
+				const hasPlaceholders = candidates.some( ( candidate ) =>
+					candidate.match( REGEXP_PLACEHOLDER )
+				);
+
+				if ( ! hasPlaceholders ) {
+					return;
+				}
+
+				const comments = context.getCommentsBefore( node ).slice();
+
+				let parentNode = parent;
+
+				/**
+				 * Loop through all parent nodes and get their preceding comments as well.
+				 *
+				 * This way we can gather comments that are not directly preceding the translation
+				 * function call, but are just on the line above it. This case is commonly supported
+				 * by string extraction tools like WP-CLI's i18n command.
+				 */
+				while (
+					parentNode &&
+					parentNode.type !== 'Program' &&
+					Math.abs( parentNode.loc.start.line - currentLine ) <= 1
+				) {
+					comments.push( ...context.getCommentsBefore( parentNode ) );
+					parentNode = parentNode.parent;
+				}
+
+				for ( const comment of comments ) {
+					const {
+						value: commentText,
+						loc: {
+							start: { line: commentLine },
+						},
+					} = comment;
+
+					/*
+					Skip cases like this:
+
+					// translators: %s: Preference
+					console.log(
+						sprintf(
+							__( 'Preference: %s' ),
+							preference
+						)
+					);
+					 */
+					if ( Math.abs( commentLine - currentLine ) > 1 ) {
+						continue;
```

Comment:
Wouldn't it be the case that if this condition is satisfied, all the rest of `comment` in `comments` would as well?

i.e.

```suggestion
					if ( Math.abs( commentLine - currentLine ) > 1 ) {
						break;
```

Reply:
I'd have to have a closer look. That would depend on the order of the comments added earlier, right?

Reply:
If it's ordered as:

- The node's immediately preceding comments
- The parent ancestry and all their preceding comments

... then, I think it would be fair to say that if we've reached the point where we're further than one line from the start of a comment, then the same would be true for all other comments in the array.

i.e. Assuming that parent nodes (and their comments) always occur at or before the node under test.